hudi-common/src/main/java/org/apache/hudi/common/config/HoodieMetadataConfig.java

```diff
@@ -398,7 +398,7 @@ public final class HoodieMetadataConfig extends HoodieConfig {
 
   public static final ConfigProperty<Boolean> SECONDARY_INDEX_ENABLE_PROP = ConfigProperty
       .key(METADATA_PREFIX + ".index.secondary.enable")
-      .defaultValue(true)
+      .defaultValue(false)
```

Comment:
hey @codope : can you comment on why this is enabled by default?

Reply:
@danny0405 : this might be stemming from the sql experience. 
for eg, 
user triggers a `create index` command via sql. 
and then adds few commits. 
exits the sql and comes back and adds new commits. 
if we were to make this flip (L401), we might expect the user to explicitly set "hoodie.meadata.index.secondary.enable" = true for sql writes. if not automatic detection and dropping of SI will happen w/ ingestion writes.

Reply:
but I also see default value for expression index is disabled. So, not sure how that would play out. 
@codope : can you help explain the user experience in these cases.

Reply:
When users create EI, they explicitly specify index type - see `using` and `expr` keywords in https://hudi.apache.org/docs/sql_ddl#create-expression-index.
But, SI creation is very simple like databases - https://hudi.apache.org/docs/sql_ddl#create-secondary-index
We can set default to false but need a way to infer that index being created is a secondary index. For EI, it's done on the basis of keywords in the create index command.